src/pulp/solvers.py

```diff
@@ -1420,7 +1420,11 @@ def solve_CBC(self, lp, use_mps=True):
         args.append(self.path)
         args.extend(cmds[1:].split())
         cbc = subprocess.Popen(args, stdout = pipe, stderr = pipe)
-        if cbc.wait() != 0:
+
+        if processTimeout is not None and sys.version_info < (3, 3):
+            warnings.warn('Pulp: Subprocess timeout is not available in python < 3.3. Solver will run indefinitely.')
+
+        if (cbc.wait(processTimeout) if sys.version_info >= (3, 3) else cbc.wait()) != 0:
```

Comment:
I would suggest this way of catching the _TimeoutExpired_ exception and also killing the cbc subprocess, as it was still able to run after the _cbc.wait(processTimeout)_.
```suggestion
        if processTimeout is not None and sys.version_info >= (3, 3):
            try:
                 cbc.wait(processTimeout)
            except subprocess.TimeoutExpired:
                cbc.kill()
                raise PulpSolverError("Pulp: Error while trying to execute " + \
                                  self.path)
        else:
            if cbc.wait()!=0:
                raise PulpSolverError("Pulp: Error while trying to execute " + \
                                      self.path)
```